lib/views/frontend/spree/checkout/payment/_applepay.html.erb

```diff
@@ -1,47 +1,33 @@
-<style>
-  #apple-pay-button {
-    display: none;
-    background-color: black;
-    background-image: -webkit-named-image(apple-pay-logo-white);
-    background-size: 100% 100%;
-    background-origin: content-box;
-    background-repeat: no-repeat;
-    width: 100%;
-    max-width: 150px;
-    height: 20px;
-    padding: 8px 0;
-    border-radius: 4px;
-  }
-  @media screen and (max-width: 767px) {
-    #apple-pay-button {
-      margin: 0 auto;
-    }
-  }
-</style>
 <span id="apple-pay-button"></span>
 
-<script type="text/javascript" src="https://js.stripe.com/v2/"></script>
 <script type="text/javascript">
-  Stripe.setPublishableKey("<%= payment_method.preferred_publishable_key %>");
-</script>
 
-<script>
-  Stripe.applePay.checkAvailability(function(available) {
-    if (available) {
-      document.getElementById('apple-pay-button').style.display = 'block';
-    } else {
-      $label = $('#payment-method-fields').find($("label:contains('apple pay')"));
-      $label.hide();
-      $label.find($("input[type=radio]")).attr('disabled', true);
-      document.getElementById('apple-pay-button').style.display = 'none';
-    }
-  });
+  if (typeof Stripe === 'undefined') {
+    $.getScript('https://js.stripe.com/v2/', applePay);
+  } else {
+    applePay();
+  };
+
+  function applePay() {
+    <% stripe_payment_method = Spree::PaymentMethod.find_by(type: 'Spree::Gateway::StripeGateway') %>
```

Comment:
@harsh-ag `<% stripe_payment_method = Spree::PaymentMethod.available_on_front_end.find_by(type: 'Spree::Gateway::StripeGateway') %>`
This will help you remove `stripe_payment_method.active?` check and also ensure that this payment method is available for frontend.